signer/tests/integration/transaction_signer.rs

```diff
@@ -161,6 +201,150 @@ async fn should_be_able_to_participate_in_signing_round() {
+                threshold: 2,
+                rng: rand::rngs::StdRng::seed_from_u64(51),
+            }
+        })
+        .collect();
+
+    // Generate test data
+    let test_params = testing::storage::model::Params {
+        num_bitcoin_blocks: 2,
+        num_stacks_blocks_per_bitcoin_block: 1,
+        num_deposit_requests_per_block: 2,
+        num_withdraw_requests_per_block: 2,
+        num_signers_per_request: 0,
+    };
+    let test_data = TestData::generate(&mut rng, &[], &test_params);
+    let bitcoin_chain_tip = test_data.bitcoin_blocks.last().unwrap().block_hash;
+
+    // Write the same test data to each signer's storage
+    for signer in signers.iter() {
+        test_data.write_to(&signer.context.get_storage_mut()).await;
+    }
+
+    // Find the coordinator signer
+    let coordinator_signer_info = signer_info
+        .iter()
+        .find(|signer| {
+            let pk = PublicKey::from_private_key(&signer.signer_private_key);
+            transaction_coordinator::given_key_is_coordinator(
+                pk,
+                &bitcoin_chain_tip,
+                signer_set_pubkeys,
+            )
+        })
+        .expect("could not determine coordinator");
+
+    // Start the event loops for each signer
+    let handles = signers
+        .into_iter()
+        .map(|signer| tokio::spawn(signer.run()))
+        .collect::<Vec<_>>();
+
+    // Give the event loops some time to start up
+    // NOTE: We could remove this kind of sleep if we had a startup-event or something.
+    tokio::time::sleep(tokio::time::Duration::from_millis(500)).await;
+
+    // Create a `SweepTransactionInfo` message and broadcast it to the network
+    let sweep_tx_info = sweep_transaction_info(
+        &mut rng,
+        *bitcoin_chain_tip,
+        &test_data.deposit_requests,
+        &test_data.withdraw_requests,
+    );
+
+    // Convert the `SweepTransactionInfo` into a `Payload` and sign it using
+    // the coordinator's keys.
+    let payload: message::Payload = sweep_tx_info.clone().into();
+    let msg = payload
+        .to_message(bitcoin_chain_tip)
+        .sign_ecdsa(&coordinator_signer_info.signer_private_key)
+        .expect("failed to sign message");
+
+    // Broadcast the message to the network
+    coord_network
+        .broadcast(msg)
+        .await
+        .expect("broadcast failed");
+
+    // Give the event loops some time to process the message
+    tokio::time::sleep(tokio::time::Duration::from_millis(500)).await;
+
+    // Ensure that the sweep transaction info has been stored in each signer's
+    // database and is the "latest" sweep transaction.
+    for db in signer_dbs.iter() {
+        let retrieved_tx = db
+            .get_latest_sweep_transaction(&bitcoin_chain_tip, 10)
+            .await
+            .expect("failed to get sweep transaction")
+            .expect("no sweep transaction found");
+
+        assert_eq!(*retrieved_tx.txid, sweep_tx_info.txid);
```

Comment:
Note - think i found a bug in the storage; you can start your review but I'll probably be pushing some small fix

Reply:
Was the test, `ON CONFLICT DO NOTHING` bit me ;) Fixed in [46ac7a1](https://github.com/stacks-network/sbtc/pull/732/commits/46ac7a10c7c820f3be5a56f8c72dc8476b715b4a)